dev/tasks/gandiva-jars/build-cpp-osx.sh

```diff
@@ -31,6 +31,8 @@ pushd cpp
           -DARROW_GANDIVA_STATIC_LIBSTDCPP=ON \
           -DARROW_BUILD_TESTS=ON \
           -DARROW_BUILD_UTILITIES=OFF \
+          -DPARQUET_BUILD_ENCRYPTION=OFF \
+          -DARROW_PARQUET=OFF \
```

Comment:
isn't this off by default?

Reply:
no encryption is on by default (not sure why).
also i want to decouple this from the default settings..